File: fast_sync_dart_client/fast_sync_client/lib/src/absraction/service/ihttp_manager.dart
import 'package:fast_sync_client/src/absraction/models/sync_operation_metadata.dart';
import 'package:fast_sync_client/src/absraction/models/sync_payload.dart';

abstract class IhttpManager {
  Future<bool> push(SyncPayload payload);
  Future<SyncPayload> pull(SyncOperationMetadata metadata);
}

File: fast-sync-typescript-client-sdk/packages/fast-sync-client/src/abstraction/service/ihttp_manager.ts
import { SyncOperationMetadata } from "../models/sync_operation_metadata";
import { SyncPayload } from "../models/sync_payload";

export interface IHttpManager {
  push(payload: SyncPayload): Promise<boolean>;
  pull(metadata: SyncOperationMetadata): Promise<SyncPayload>;
}
